my-dsl.md

```diff
@@ -3,37 +3,112 @@
+this DSL. 
+
+It is also very domain-specific because there is very little you can do with
+this DSL outside of representing sheet-music; it is most certainly not
+Turing-complete. It is specifically focused on generating a parseable string
+which can be fed to other music programs.
+
 # Internal or external?
 
 _Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+This is an external DSL because it is written in its own distinct language,
+which can then be processed by other languages. In this case, the notation that
+the DSL produces can be fed to other music programming languages such as Lilpond
+and ABC notation.
+
 # Host language
 
 _What language(s) was (were) used to implement the DSL?_
 
+Lilypond and/or ABC Notation which are both used to generate sheet music PDFs.
+It can also be fed into other languages for deep learning tasks, such as OMR. 
+
 # Benefits
 
 _Identify one potential benefit of the DSL: how is a programmer's life made
 easier by the existence of this language?_
 
+From my understanding, the idea is that rewriting sheet music code can be
+cumbersome. Suppose you have sheet music which has multiple voices, and that you
+wanted to add a repeat sign to one of the staffs. Then, you'd also want the
+other voices to have that repeat sign as well. It seems like this is painful to do
+in the current music languages that we have because they require that you code
+each different voice separately. So the edits you do to one voice will not apply
+to another.
+
+The DSL circumvents this by breaking up the music into measures that you can
+easily manipulate, instead of having you edit each voice/staff. Let's go back to
+the previous example where we have some music with 3 measures and 2 voices,
+soprano and alto. We want to add a repeat sign to the end of the measure 2 for
+the soprano voice. Without the DSL, we would have to go to the section of code
+that refers to the soprano voice, add the repeat sign in, then go to the alto
+voice, and add the repeat sign again. But with the DSL, we can use the notation
+2 * [1,2], 3 to indicate that we want to take all the music at measures 1 and 2
+and repeat it twice before continuing onto measure 3. 
+
 # Drawbacks
 
 _Identify one potential drawback of the DSL: what does a programmer
 lose by using this DSL instead of a general-purpose language?_
+
+The notation is very regex like; that is to say, you need to understand
+different symbols in order to unpack the string you are reading/want to
+write. Thankfully, the author tries to call on past conventions to make this
+transition easier (Python also has the notation of multiplying a constant by an
+array to represent repetition, like 2 * [1, 2]) but this requires some coding
+knowledge. So in that sense, it may be inconvenient to learn a new language,
+therefore making the programmer want to rely on a GPL that is more familiar to
+them. 
```

Comment:
I feel like since this language is fairly simple, the time learning it would be insignificant compared to the time saved by using it if you program sheet music regularly.